src/AstTypeAnalysis.cpp

```diff
@@ -509,8 +509,8 @@ std::map<const AstArgument*, TypeSet> TypeAnalysis::analyseTypes(
                 }
 
                 // add constraints for arguments
-                for (size_t i = 0; i < fun.getArity(); i++) {
-                    auto arg = getVar(fun.getArg(i));
+                for (size_t i = 0; i < fun.getArguments().size(); i++) {
```

Comment:
can you change this loop to size_t i=0; for(auto arg: fun.getArguments()) { ....; i++}